### PR TITLE
[pr] Borsh Serialize solana_sdk::Transaction

### PR DESCRIPTION
**Pull request by DudessaPr**
#### Problem
In [svm-rollup pr](https://github.com/nitro-svm/svm-rollup/pull/76) we are using Transaction inside of CallMessage, which requires Borsh

#### Summary of Changes

Added `BroshSerialization/Deserialization` to `Transaction` struct, and other structs inside of it, such as `Signature`, `CompiledInstruction`, etc.

Fixes #

### CODE REVIEW

File: sdk/src/signature.rs [hunk 1]
```diff
@@ -25,6 +28,49 @@ const MAX_BASE58_SIGNATURE_LEN: usize = 88;
...
+        any::<[u8; 64]>()
+            .prop_map(|bytes| Signature(GenericArray::clone_from_slice(&bytes)))
+            .boxed()
+    }
+
+    type Strategy = proptest::prelude::BoxedStrategy<Signature>;
+}
+
+// Implement Arbitrary for Signature
+impl<'a> arbitrary::Arbitrary<'a> for Signature {
```
Comment (jason-nitro) [inline sdk/src/signature.rs:44]: Just a Rust language question.. was the lifetime annotation required here? I would've thought this was a case where it was just implicit.

Comment (DudessaPr) [reply sdk/src/signature.rs:44]: It was required by trait definition of arbitrary::Arbitrary

**Comment by Yiwen-Gao** [comment]
Can you elaborate on why Arbitrary and Proptest are needed? It'd be helpful to link code snippets as well.

**Comment by Yiwen-Gao** [comment]
They're testing libraries, so even if we do need them, I believe they should be gated behind configuration flags.

**Comment by DudessaPr** [comment]
> Can you elaborate on why Arbitrary and Proptest are needed? It'd be helpful to link code snippets as well.

I see some sovereign modules use Arbitrary and Proptest for fuzz testing. And in the module template we had those created derived, I mean they had the following in callMessage
```
#[cfg_attr(
    feature = "arbitrary",
    derive(arbitrary::Arbitrary, proptest_derive::Arbitrary)
)]
```

I believe we will need those crates in the future for more testing to find bugs in the module. Therefore, decided not to remove it from CallMessage and implement it for the Transaction type.

**Comment by DudessaPr** [comment]
> They're testing libraries, so even if we do need them, I believe they should be gated behind configuration flags.

You are right, I missed adding flags here. I did not think about them and totally forgot. Will change that

**Review by Yiwen-Gao — APPROVED** [review]

**Pull request merged**